pkg/console/operator/sync_v400.go

```diff
@@ -408,3 +410,31 @@ func getDeploymentGeneration(co *consoleOperator) int64 {
 	}
 	return deployment.Generation
 }
+
+func copyCustomLogoConfigMap(co *consoleOperator, operatorConfig *operatorv1.Console) {
+	// Check if a custom logo is set on the operator config
+	logoName := operatorConfig.Spec.Customization.CustomLogoFile.Name
+	if logoName != "" {
+		// If it exist, copy the configMap in openshift-config into openshift-console
+		customLogoConfigMap, logoErr := co.configMapClient.ConfigMaps(api.OpenShiftConfigNamespace).Get(logoName, metav1.GetOptions{})
+		if logoErr != nil && !apierrors.IsNotFound(logoErr) {
+			logrus.Errorf("custom logo configMap error: %v \n", logoErr)
+		} else if apierrors.IsNotFound(logoErr) {
+			logrus.Errorf("warning: custom logo config map name is set but configmap is not found %v \n", logoErr)
+		} else {
+			customLogoConfigMap.Namespace = "openshift-console"
```

Comment:
Can you choose to do something like this instead:

```golang
	logoConfigMap := &corev1.ConfigMap{
		ObjectMeta: metav1.ObjectMeta{
                         // We have to honor the ConfigMapNameRef in our operator config, 
                         // but we can copy it into `openshift-console` and name it something stable.
                         //  Could this simplify some of the logic?
			Name:     "user-custom-logo-configmap",
			Namespace: Namespace:   api.OpenShiftConsoleNamespace,
		},
                 // copy over the data section instead of manipulating the original?
		Data: OriginalConfigMapFromOpenShiftConfig.Data
	}
```

Reply:
I still think whenever you resort to clearing the resource version:

```golang
customLogoConfigMap.ResourceVersion = ""
```
the approach is incorrect.